docs/reference/modules/transport.asciidoc

```diff
@@ -20,42 +20,90 @@ as search in Elasticsearch.
+`transport.port`::
+A bind port range. Defaults to `9300-9400`.
 
-|`transport.publish_port` |The port that other nodes in the cluster
+`transport.publish_port`::
+The port that other nodes in the cluster
 should use when communicating with this node. Useful when a cluster node
 is behind a proxy or firewall and the `transport.port` is not directly
 addressable from the outside. Defaults to the actual port assigned via
 `transport.port`.
 
-|`transport.bind_host` |The host address to bind the transport service to. Defaults to `transport.host` (if set) or `network.bind_host`.
+`transport.bind_host`::
+The host address to bind the transport service to. Defaults to
+`transport.host` (if set) or `network.bind_host`.
 
-|`transport.publish_host` |The host address to publish for nodes in the cluster to connect to. Defaults to `transport.host` (if set) or `network.publish_host`.
+`transport.publish_host`::
+The host address to publish for nodes in the cluster to connect to.
+Defaults to `transport.host` (if set) or `network.publish_host`.
 
-|`transport.host` |Used to set the `transport.bind_host` and the `transport.publish_host`.
+`transport.host`::
+Used to set the `transport.bind_host` and the `transport.publish_host`.
 
-
-|`transport.connect_timeout` |The connect timeout for initiating a new connection (in
+`transport.connect_timeout`::
+The connect timeout for initiating a new connection (in
 time setting format). Defaults to `30s`.
 
-|`transport.compress` |Set to `true` to enable compression (`DEFLATE`) between
+`transport.compress`::
+Set to `true` to enable compression (`DEFLATE`) between
 all nodes. Defaults to `false`.
 
-|`transport.ping_schedule` | Schedule a regular application-level ping message
+`transport.ping_schedule`::
+Schedule a regular application-level ping message
 to ensure that transport connections between nodes are kept alive. Defaults to
 `5s` in the transport client and `-1` (disabled) elsewhere. It is preferable
 to correctly configure TCP keep-alives instead of using this feature, because
 TCP keep-alives apply to all kinds of long-lived connections and not just to
 transport connections.
 
-|=======================================================================
-
 It also uses the common
 <<modules-network,network settings>>.
 
+`transport.tcp.no_delay`::
+Enable or disable the https://en.wikipedia.org/wiki/Nagle%27s_algorithm[TCP no delay]
+setting. Defaults to `network.tcp.no_delay`.
+
+`transport.tcp.keep_alive`::
+Configures the `SO_KEEPALIVE` option for this socket, which
+determines whether it sends TCP keepalive probes.
+Defaults to `network.tcp.keep_alive`.
+
+`transport.tcp.keep_idle`::
+Configures the `TCP_KEEPIDLE` option for this socket, which
+determines the time in seconds that a connection must be idle before
+starting to send TCP keepalive probes. Defaults to `network.tcp.keep_idle`, which
+is the system default. May not be greater than 300. In case where the system default
```

Comment:
This is awkward to explain; my best idea so far is this:

```suggestion
Defaults to `network.tcp.keep_idle` if set, or the system default otherwise.
May not be greater than 300. In case where the system default
```

Maybe the docs folks can think of a better way to explain this?